heidrun/nara_json.rb

```diff
@@ -0,0 +1,575 @@
+def nara_catalog_uri(id)
+  "http://catalog.archives.gov/id/#{id.node}"
+end
+
+# See object and preview mappings
+def make_obj_uri(obj)
+  URI::escape(obj.node['@url'])  # Can have space characters
+end
+def make_obj_dcformat(obj)
+  obj.node['@mime']
+end
+def make_preview_uri(obj)
+  URI::escape(obj.node['@url'])  # Can also have space characters
+end
+def make_preview_dcformat(obj)
+  obj.node['@mime']
+end
+
+# Return a string suitable for sourceResource.contributor or
+# sourceResource.publisher.
+#
+# @see #make_contributor
+# @see #make_publisher
+# @param contributor_array [Krikri::JsonParser::Value]
+# @return [String, RDF::Literal]
+def contributor_term_name(contributor_array)
+  node = contributor_array.node
+  contributors = node.fetch('organizationalContributor',
+                            node['personalContributor'])
+  contributors = [contributors] unless contributors.is_a? Array
+
+  yield(contributors)
+
+  return contributors.first['contributor']['termName'] \
+    unless contributors.empty?
+
+  # FIXME:  It's not clear how to indicate the concept of "nothing" to
```

Comment:
I definitely want to talk as a group about it -- especially with @guegueng and @AudreyAltman, who have not dealt yet with this mapping (and @guegueng not having as much of a Ruby background) because this is overwhelmingly a Ruby programming experience.  If I need detailed knowledge of how Ruby array methods work, how procs work, and so forth, then it seems I should follow various Ruby coding guidelines, like breaking things up into smaller methods.

The other thing with regard to the goals of the mappings is that I've had little documentation to go by, and I don't know where this goal is stated that it might guide my work.